### PR TITLE
Added Resources Rights tests and check security

### PR DESCRIPTION
**Pull request by mickaelandrieu**
* Added unit tests (upload success, upload too big file fails)
* Check authorizations

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
@eric-chau can you review please ? :)

File: Event/Listener/ExceptionListener.php [hunk 1]
```diff
@@ -54,6 +54,7 @@ public function __construct(BBApplication $application)
     public function onKernelException(GetResponseForExceptionEvent $event)
     {
         $exception = $event->getException();
+        var_dump($exception->getMessage());
```
Comment (eric-chau) [inline Event/Listener/ExceptionListener.php]: Don't forget to remove this ``var_dump`` =)

File: Rest/Tests/Controller/ResourceControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+            );
+
+        $request = new Request([], [], [], [], ['file' => $image], [
+            'REQUEST_URI'    => $url,
+            'CONTENT_TYPE'   => 'image/png',
+            'REQUEST_METHOD' => 'POST',
+        ]);
+
+        $response = $this->sendRequest($request);
+        $this->assertEquals(Response::HTTP_CREATED, $response->getStatusCode(), sprintf('HTTP %s expected, HTTP %s returned.', Response::HTTP_CREATED, $response->getStatusCode()));
```
Comment (eric-chau) [inline Rest/Tests/Controller/ResourceControllerTest.php]: Every line should contains 120 characters max. Do something like this please:

```php
 $this->assertEquals(
    Response::HTTP_CREATED, 
    $response->getStatusCode(), 
    sprintf('HTTP %s expected, HTTP %s returned.', Response::HTTP_CREATED, $response->getStatusCode())
);
```

**Comment by eric-chau** [comment]
Great contribution @mickaelandrieu, thanks! :+1:

File: Event/Listener/ExceptionListener.php (hunk 1 above)
Comment (mickaelandrieu) [inline Event/Listener/ExceptionListener.php]: oups ! sry

File: Rest/EventListener/ExceptionListener.php [hunk 3]
```diff
@@ -70,7 +71,7 @@ public function onKernelException(GetResponseForExceptionEvent $event)
 
             $event->getResponse()->setStatusCode($code, $message);
             $event->getResponse()->headers->add(array('Content-Type' => 'application/json'));
-        } elseif ($exception instanceof HttpExceptionInterface) {
+        } elseif ($exception instanceof HttpExceptionInterface || $exception instanceof HttpException) {
```
Comment (crouillon) [inline Rest/EventListener/ExceptionListener.php]: Why is need this test? HttpException implementing HttpExceptionInterface it should be always the same result for both tests

Comment (mickaelandrieu) [inline Rest/EventListener/ExceptionListener.php]: OMG :/ removed.

**Comment by mickaelandrieu** [comment]
@eric-chau comments adressed, thanks for review :)

**Pull request merged**